src/transformStateWithClones.js

```diff
@@ -7,7 +7,30 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateHistory = [];
+  let stateClone = Object.assign({}, state);
+
+  actions.forEach(function (element) {
+    if (element.type === 'clear') {
+      stateClone = {};
+      stateHistory.push({ ...stateClone });
```

Comment:
According to the checklist, you should avoid repeating the same action in all switch cases. Here, pushing the stateClone to the stateHistory is repeated in each case. You should perform this action just once after the current action is processed, at the end of the loop cycle but outside of the switch block.